src/components/application_manager/src/hmi_language_handler.cc

```diff
@@ -288,11 +290,14 @@ void HMILanguageHandler::HandleWrongLanguageApp(const Apps::value_type& app) {
                       << " is not found within apps with wrong language.");
     return;
   }
+  apps_.erase(it);
+  if (0 == apps_.size()) {
+    LOG4CXX_DEBUG(logger_,
```

Comment:
@dtrunov 
Logging is ambiguous - at least last application will be processed right after that message.